platforms/ide/problems-rendering/src/main/java/org/gradle/problems/internal/rendering/ProblemRenderer.java

```diff
@@ -45,32 +49,61 @@ public void render(InternalProblem problem) {
     }
 
     private static void render(PrintWriter output, List<InternalProblem> problems) {
-        String sep = "";
-        for (InternalProblem problem : problems) {
-            output.printf(sep);
-            renderProblem(output, problem);
-            sep = "%n";
+        // Group problems by problem id
+        // When generic rendering is addressed, maybe we also group by the whole problem group hierarchy
+        Map<ProblemId, List<InternalProblem>> problemIdListMap = problems.stream().collect(Collectors.groupingBy(internalProblem -> internalProblem.getDefinition().getId()));
+        String separator = "";
+        for (Map.Entry<ProblemId, List<InternalProblem>> problemIdListEntry : problemIdListMap.entrySet()) {
+            renderProblemsById(output, problemIdListEntry.getKey(), problemIdListEntry.getValue(), separator);
+            separator = "%n";
         }
     }
 
-    static void renderProblem(PrintWriter output, InternalProblem problem) {
-        boolean isJavaCompilationProblem = problem.getDefinition().getId().getGroup().equals(GradleCoreProblemGroup.compilation().java());
+    private static void renderProblemsById(PrintWriter output, ProblemId problemId, List<InternalProblem> problems, String separator) {
+        String sep = separator;
+        boolean isJavaCompilationProblem = problemId.getGroup().equals(GradleCoreProblemGroup.compilation().java()) && !problemId.getName().equals("initialization-failed");
```

Comment:
Perhaps I sound like a broken record here, but

This feels very weird to me -- that our general purpose problem reporting infrastructure having _special_ handling of java errors, and hard-coding certain constants like `initialization-failed`.

Reply:
I think this definitely deserves a TODO. 

Long term I really think this infrastructure should be as generic as possible